include/aspect/assemblers.h

```diff
@@ -0,0 +1,106 @@
+/*
+  Copyright (C) 2016 by the authors of the ASPECT code.
```

Comment:
Let's just discuss the separation of files. What if you just separated out the *implementation* of assemblers into their own files -- say into `assembly_stokes.cc` and `assembly_advection.cc` but kept with a single header file?